securedrop/journalist_app/__init__.py

```diff
@@ -110,14 +110,10 @@ def _handle_http_exception(
     def expire_blacklisted_tokens() -> None:
         cleanup_expired_revoked_tokens()
 
-    @app.before_request
-    def load_instance_config() -> None:
-        app.instance_config = InstanceConfig.get_current()
```

Comment:
Actually the instance config is now retrieved every time InstanceConfig.get_default() is called, so it's  actually hitting the DB as much as before (or potentially more often if a single request has multiple .get_default()s) This reduces the benefit of a global object, so there's probably some opportunity for optimization there. The trick would be to detect configuration changes when they're made in the journalist interface, as we want those to be reflected immediately in the source app.

Reply:
Updated the InstanceConfig get_default() method to be lazier - it will now only hit the db if explicitly requested or if the global is `None`. With some tweaks to `@app.before_request`-decorate functions, changes should show up immediately in both apps without extra DB overhead.

Also, re `valid_until` - it might be a bit counter-intuitive. It records the historical info about previous configs, and when they were invalidated. (TBH I'm not sure we need to store said previous configs, but it does provide an audit trail for instance_config changes.)